### PR TITLE
Move support to v2.7.x and v3.1.0

### PR DESCRIPTION
**Pull request by pablogs9**
Signed-off-by: Pablo Garrido <pablogs9@gmail.com>

### CODE REVIEW

**Comment by uLipe** [comment]
@pablogs9 as I said [here](https://github.com/micro-ROS/micro_ros_zephyr_module/pull/63), to get this working we would need to update some points:

* USB-Serial now used device tree to get enabled, but applying this update will break the compatibillity with previous Zephyr releases;
* Wait ros2 accept the rcuitls patch, for the posix timer implementation and update the commit used by libmicroros, currently I created a diff and patching locally and wrote this instruction in the app I'm working on top of this module.

Also I'm taking some time to refactor the zephyr transports for serial to use the up to date DT macros instead of using device labels, this is compatible across zephyr releases and scalable as long Zephyr evolves, probably will push this PR today or tomorrow.

**Comment by uLipe** [comment]
Sure, I will take some time to look into :)

**Comment by pablogs9** [comment]
@mergify backport humble galactic foxy

File: modules/libmicroros/microros_transports/serial-usb/microros_transports.c [hunk 1]
```diff
@@ -58,12 +58,15 @@ bool zephyr_transport_open(struct uxrCustomTransport * transport){
     int ret;
     uint32_t baudrate, dtr = 0U;
 
-    params->uart_dev = device_get_binding("CDC_ACM_0");
+
+    // params->uart_dev = device_get_binding("CDC_ACM_0");
+    params->uart_dev = DEVICE_DT_GET_ONE(zephyr_cdc_acm_uart);
+    printk("uart_dev: %p\n", params->uart_dev);
```
Comment (pablogs9) [inline modules/libmicroros/microros_transports/serial-usb/microros_transports.c:64]:
```suggestion
```

File: modules/libmicroros/microros_transports/serial-usb/microros_transports.c [hunk 2]
```diff
@@ -58,12 +58,15 @@ bool zephyr_transport_open(struct uxrCustomTransport * transport){
...
     if (!params->uart_dev) {
         printk("CDC ACM device not found\n");
         return false;
     }
-
+    printk("CDC ACM device found: ready: %d\n", device_is_ready(params->uart_dev));
```
Comment (pablogs9) [inline modules/libmicroros/microros_transports/serial-usb/microros_transports.c:69]:
```suggestion
```

File: modules/libmicroros/microros_transports/serial-usb/microros_transports.c (hunk 1 above)
Comment (pablogs9) [inline modules/libmicroros/microros_transports/serial-usb/microros_transports.c:64]:
```suggestion
    params->uart_dev = device_get_binding("CDC_ACM_0");
```

**Comment by Acuadros95** [comment]
FYI @pablogs9 @uLipe , we need to change zephyr include paths for `v.3.1.0` and beyond:

https://github.com/zephyrproject-rtos/zephyr/releases/tag/zephyr-v3.1.0:
> All Zephyr public headers have been moved to include/zephyr, meaning they
need to be prefixed with <zephyr/...> when included. Because this change
can potentially break many applications or libraries,
:kconfig:option:CONFIG_LEGACY_INCLUDE_PATH is provided to allow using the
old include path. This option is now enabled by default to allow a smooth
transition. In order to facilitate the migration to the new include prefix, a
script to automate the process is also provided:
:zephyr_file:scripts/utils/migrate_includes.py.

**Comment by uLipe** [comment]
Yes, altough this could be done in other PR, the kconfig option for legacy
includes is only disabled to run apps under twister which is used to do
unit tests for zephyr components.


Le mer. 17 août 2022 à 07:45, Antonio Cuadros ***@***.***> a
écrit :

**Comment by pablogs9** [comment]
Do we have any define in the Zephyr build system indicating the version so we can conditionally include the correct header?

**Pull request merged**